core/src/main/java/org/elasticsearch/common/geo/GeoPoint.java

```diff
@@ -126,14 +126,10 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
```

Comment:
can we have a simple unittest that equals and hashcode don't violate it's contract? ie if `p.equals(q) then p.hashCode() == q.hashCode()`